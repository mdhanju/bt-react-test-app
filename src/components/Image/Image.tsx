import React from 'react'
import classnames from 'classnames'

const Image = ({
    className,
    src,
    alt,
    ...props
}: Props): React.ReactElement => (
    <img src={src} className={classnames(className)} alt={alt} {...props} />
)

type Props = {
    className?: string;
    src: string;
    alt?: string;
};

export default Image
