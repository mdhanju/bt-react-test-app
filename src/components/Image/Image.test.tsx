import React from 'react'

import Image from './Image'
import desktopLogoSrc from '../../assets/images/logo-white.png'

describe('<Image />', () => {
    const defaultProps = {
        onClick: jest.fn()
    }

    it('should render properly without crashing', () => {
        const wrapper = shallow(<Image {...defaultProps} />)
        expect(wrapper.exists()).toBe(true)
        expect(wrapper.getElements()).toMatchSnapshot()
    })

    it('should respond to passed in function when clicked', () => {
        const wrapper = shallow(<Image {...defaultProps} />)
        wrapper.simulate('click')

        expect(defaultProps.onClick).toHaveBeenCalled()
    })
    //TODO
    it.skip('render the image in html', () => {
        const wrapper = render(
            <Image
                src={desktopLogoSrc}
                className="mobile-hide"
                alt="brand logo"
            />
        )
        expect(wrapper.html()).toMatch(
            '<img src="logo-white.png" class="mobile-hide">'
        )
    })
})
