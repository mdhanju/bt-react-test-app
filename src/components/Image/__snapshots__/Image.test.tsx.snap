// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Image /> should render properly without crashing 1`] = `
Array [
  <img
    className=""
    onClick={[MockFunction]}
  />,
]
`;
